solvers/skglm.py

```diff
@@ -4,15 +4,18 @@
 
 with safe_import_context() as import_ctx:
     import numpy as np
+    from scipy import sparse
     from skglm.datafits import Quadratic, Quadratic_32
     from skglm.penalties import L1
     from skglm.solvers import cd_solver_path
     from sklearn.exceptions import ConvergenceWarning
+    from sklearn.linear_model._base import _preprocess_data
 
 
 class Solver(BaseSolver):
     name = "skglm"
     stopping_strategy = "iteration"
+    support_sparse = False
```

Comment:
it does support sparse, it's just that it does not support sparse and fit_intercept at the same time